crates/ra_fmt2/src/fmt_diff.rs

```diff
@@ -16,10 +16,90 @@ use std::rc::Rc;
 #[derive(Debug, Clone)]
 pub(crate) struct DiffView {
     // TODO more diffing info not just a string
-    diffs: Vec<SmolStr>,
+    diffs: Vec<(usize, SmolStr)>,
     original: SmolStr,
 }
 
+impl DiffView {
```

Comment:
Is modifying the string directly ok? It much easier that messing with them as blocks still.

Reply:
I think it's definitely OK to start with